### PR TITLE
Update the command line interface

### PR DESCRIPTION
**Pull request by neganp**
This PR changes the command line interface of Nucamino to an updated interface, as discussed in #3. This comes with a number of other changes:

- Migrate from `go-flags` to `cobra`; `cobra` is a good fit for the new CLI structure, and is used by a number of major Go projects (Kubernetes, Hugo, rkt, etcd, docker, etcd), and is decently documented and maintained.
- Update the `hack/` scripts so as to not break the Makefile /  continuous integration.
- Start adding unit tests to the `cli` and `cmd` modules.
- Update the `README` to reflect the new CLI.

While re-structuring the CLI, a `main` module was added running `go build .` in a properly configured go work-space will now build nucamino without having to install docker. The test and profiling scripts can also be run without a docker environment (though cross-compilation may require downloading addition Go runtime components, which must be done a root, which is probably something that merits a container).

The new CLI adds the ability to load an alignment profile at execution time, as well as a sub-command for listing and printing built-in profiles, as well as validating custom profiles.

Built-in alignment profiles were moved to their own module; its only responsibilities are holding, listing, and retrieving built-in profiles. To avoid parsing built-in profiles at runtime, they're stored as Go source code rather than embedded data (the equivalent YAML can be retrieved with the `profile print` command).

Additional minor changes:

- Fixed some formatting (with `gofmt`).
- Validate the `output-format` flag before performing alignment (`cobra` doesn't restrict input flags by value).

### CODE REVIEW

**Comment by philiptzou** [comment]
@neganp I'm going to merge this pull request once the two review comments get solved. Thanks!

**Comment by neganp** [comment]
Glad to hear it :+1: ! 

I don't see any review comments, though?

File: cmd/align.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+var alignLongMsg = `
+Loads nucleotide sequences from a FASTA file and aligns them using a
+built-in profile. The first argument is the name of the built-in
+profile to use for the alignment. The second argument is a comma
+separated list of genes to align against. (This list should either be
+surrounded by quote marks or contain no spaces).
+
+Examples:
+
+	nucamino align hiv1b pol
```
[the remark this answers is not in the document]
Comment (neganp) [reply cmd/align.go]: Good catch! Fixed. 

There isn't currently a way to see available genes other than looking at the keys in the ReferenceSequences dict in a profile (by doing `nucamino profile print $profile_name`).

I could split `list` into `list-profiles` and `list-genes $profile_name`?

**Comment by philiptzou** [comment]
I think `nucamino profile list-genes $profile_name` is a good idea. But I'll merge this one first.

**Pull request merged**